### PR TITLE
[Ide] Added a disposed token to the document controller API.

### PR DESCRIPTION
**Pull request by mkrueger**
The disposed token can be used by continue with calls that shouldn't
continue on disposed instances.

### CODE REVIEW

**Comment by mkrueger** [comment]
Set the DisposedToken in constructor, because the token source gets set to null in Dispose.

**Comment by Therzok** [comment]
I'm okay with having disposed surfaced as protected.

**Comment by sevoku** [comment]
+1 for it to be protected, and maybe ad a `public bool IsDisposed`?

File: main/src/addins/MonoDevelop.TextEditor/MonoDevelop.TextEditor/TextViewContent.cs [hunk 1]
```diff
@@ -841,4 +838,3 @@ public void WriteTextTo (TextWriter writer, int offset, int length)
 				=> throw new NotImplementedException ();
 		}
 	}
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.TextEditor/MonoDevelop.TextEditor/TextViewContent.cs]:
```suggestion
	}
}
```

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Documents/DocumentController.cs [hunk 2]
```diff
@@ -384,12 +384,26 @@ public abstract class DocumentController : IDisposable, ICommandDelegatorRouter
 			}
 		}
 
+		CancellationTokenSource disposedTokenSource = new CancellationTokenSource ();
+
+		/// <summary>
+		/// Returns a dispose token that's canceled when the document controller disposes.
+		/// </summary>
+		protected CancellationToken DisposedToken { get; private set; }
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Documents/DocumentController.cs]:
```suggestion
		protected CancellationToken DisposedToken { get; }
```

We never change it, so let's not set it anywhere else except the constructor.

**Comment by Therzok** [comment]
@mkrueger can you fix the build?

**Pull request merged**